syncer/checkpoint.go

```diff
@@ -106,21 +119,23 @@ func (b *binlogPoint) rollback() (isSchemaChanged bool) {
 func (b *binlogPoint) outOfDate() bool {
 	b.RLock()
 	defer b.RUnlock()
-	return binlog.ComparePosition(b.Position, b.flushedPos) > 0
+	// TODO: add gtid compare
+	//return pbinlog.ComparePosition(b.Position, b.flushedPos) > 0
+	return pbinlog.CompareLocation(b.location, b.flushedLocation) > 0
 }
 
-// MySQLPos returns point as mysql.Position
-func (b *binlogPoint) MySQLPos() mysql.Position {
+// MySQLPoint returns point as pbinlog.Location
+func (b *binlogPoint) MySQLLocation() pbinlog.Location {
 	b.RLock()
 	defer b.RUnlock()
-	return b.Position
+	return b.location
```

Comment:
Is `b.location.Clone` better? if so, ditto for L138.

Reply:
update in 6d3ab39 and ffeb7eb2e226d811f3fc84f152874b8c368b822a